commcare_connect/opportunity/models.py

```diff
@@ -684,6 +685,40 @@ def __setattr__(self, name, value):
                 self.status_modified_date = now()
         super().__setattr__(name, value)
 
+    def save(self, **kwargs):
+        # The code below automatically switches/create CompletedWork when the UNIQUE_USER_VISIT_CONSTRAINT
+        # is triggered.
+        rank = kwargs.pop("rank", 0)
```

Comment:
nit: can this just be handled with a default arg? or is the `pop` necessary for the `super` call